File: Go/ExampleTests/production/orderitem.go
package domain

type OrderItem struct {
	Name SimpleText
	Quantity int
	Price Dollar
	ItemTotal Dollar
}

func NewOrderItem(name SimpleText, quantity int, price Dollar, itemtotal Dollar) *OrderItem {
	return &OrderItem{Name: name, Quantity: quantity, Price: price, ItemTotal: itemtotal}
}

File: TypeScript/ExampleTests/production/OrderItem.ts
export class OrderItem {
  name: SimpleText;
  quantity: number;
  price: Dollar;
  itemTotal: Dollar;

  constructor(name: SimpleText = No Name, quantity: number = 1, price: Dollar = 1, itemTotal: Dollar = 1) {
    this.name = name;
    this.quantity = quantity;
    this.price = price;
    this.itemTotal = itemTotal;
  }
}
